app/src/main/java/com/cp/tasky/ui/sharecomponent/AgendaItemSummary.kt

```diff
@@ -0,0 +1,189 @@
+package com.cp.tasky.ui.sharecomponent
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.RadioButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.ConstraintSet
+import com.cp.tasky.R
+import com.cp.tasky.ui.theme.MEDIUM_8_DP
+import com.cp.tasky.ui.theme.SMALL_4_DP
+
+@Composable
+fun AgendaItemSummary(
+    modifier: Modifier = Modifier,
+    isDone: Boolean = false,
+    itemDescription: String = "",
+    dropdownMenuExpanded: Boolean = true,
+    itemTitle: String,
+    remindAt: String,
+    backgroundColor: Color,
+    onRadioButtonClick: (Boolean) -> Unit,
+    onDropdownMenuClick: (Boolean) -> Unit,
+    onItemOpenClick: () -> Unit,
+    onItemEditClick: () -> Unit,
+    onItemDeleteClick: () -> Unit
+) {
+
+    val constraint = ConstraintSet {
+        val (radioButton, itemTitleText, itemDescriptionText, icon) = createRefsFor(
+            RADIO_BUTTON,
+            ITEM_TITLE_TEXT,
+            ITEM_DESCRIPTION_TEXT,
+            ICON
+        )
+
+        constrain(radioButton) {
+            start.linkTo(parent.start)
+            top.linkTo(parent.top)
+        }
+
+        constrain(itemTitleText) {
+            start.linkTo(radioButton.end)
+            top.linkTo(radioButton.top)
+            bottom.linkTo(radioButton.bottom)
+        }
+
+        constrain(itemDescriptionText) {
+            start.linkTo(itemTitleText.start)
+            top.linkTo(itemTitleText.bottom)
+        }
+
+        constrain(icon) {
+            top.linkTo(itemTitleText.top)
+            bottom.linkTo(itemTitleText.bottom)
+            end.linkTo(parent.end)
+        }
+    }
```

Comment:
I'm personally not a big friend of ConstraintLayout for Compose, since you can't easily tell how the UI component looks like just from reading the code. That was better with XML's visual constraints, but for Compose the typical Column/Row/Box structure is usually much easier to understand, especially for a rather simple UI component like this agenda item